app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -116,6 +116,28 @@ object AppRepository {
             .flowOn(executionContext.database)
     }
 
+    private val refreshChangedSessionsSignal = MutableSharedFlow<Unit>()
+
+    private fun refreshChangedSessions() {
+        logging.d(javaClass.simpleName, "Refreshing changed sessions ...")
+        val requestIdentifier = "refreshChangedSessions"
+        parentJobs[requestIdentifier] = databaseScope.launchNamed(requestIdentifier) {
+            refreshChangedSessionsSignal.emit(Unit)
+        }
+    }
+
+    /**
+     * Emits all sessions from the database which have been marked as changed, cancelled or new.
+     * The returned list might be empty.
+     */
+    @OptIn(ExperimentalCoroutinesApi::class)
+    val changedSessions: Flow<List<Session>> by lazy {
+        refreshChangedSessionsSignal
+            .onStart { emit(Unit) }
+            .mapLatest { loadChangedSessions() }
+            .flowOn(executionContext.database)
+    }
```

Comment:
This is a receptive pattern here right? -> I wondering if it could be dried out.

Reply:
:thinking: Can you please clarify what you mean?